File: Leetcode/2062/2062.go
func check(runes []rune, l int, r int) bool {
	const N = 5
	stats := [N]int{}
	for i := l; i <= r; i++ {
		switch runes[i] {
		case 'a':
			stats[0]++
		case 'e':
			stats[1]++
		case 'i':
			stats[2]++
		case 'o':
			stats[3]++
		case 'u':
			stats[4]++
		default:
			return false
		}
	}
    for j := 0; j < N; j++ {
        if stats[j] == 0 {
            return false
        }
    }
	return true
}
func countVowelSubstrings(word string) int {
	runes := []rune(word)
	counter := 0
	for i := 0; i < len(runes); i++ {
		for j := i; j < len(runes); j++ {
			if check(runes, i, j) {
				counter++
			}
		}
	}
	return counter
}

File: Leetcode/2062/2062.cpp
constexpr int MAXN = 26;

class Solution {
public:
    bool check(string&word, int l, int r) {
        int stats[MAXN]{};        
        for (int i = l; i <= r; ++i) {
            if (word[i] != 'a' && word[i] != 'e' && word[i] != 'i' && word[i] != 'o' && word[i] != 'u') return false;
            ++stats[word[i]-'a'];
        }
        return stats['a'-'a'] > 0 && stats['e'-'a'] > 0 && stats['i'-'a'] > 0 && stats['o'-'a'] > 0 && stats['u'-'a'] > 0;
    }
    int countVowelSubstrings(string word) {
        int n = (int)word.size();
        int counter = 0;
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                if (check(word, i, j)) ++counter;
            }
        }
        return counter;
    }
};